warehouse/query-core/src/main/java/datawave/query/jexl/DatawavePartialInterpreter.java

```diff
@@ -0,0 +1,887 @@
+        Object right = node.jjtGetChild(1).jjtAccept(this, data);
+        try {
+            // we could have an expression like '1+1+1 == 3'
+            if (left instanceof State) {
+                State state = (State) left;
+                if (state.hasNumeric()) {
+                    left = state.getNumeric();
+                }
+            }
+            
+            result = arithmetic.equals(left, right) ? Boolean.TRUE : Boolean.FALSE;
+        } catch (ArithmeticException xrt) {
+            throw new JexlException(node, "== error", xrt);
+        }
+        
+        resultMap.put(nodeString, result);
+        return getState(node, result);
+    }
+    
+    @Override
+    public Object visit(ASTERNode node, Object data) {
+        Object match = super.visit(node, data);
+        return getState(node, match);
+    }
+    
+    @Override
+    public Object visit(ASTOrNode node, Object data) {
+        Deque<JexlNode> children = new ArrayDeque<>();
+        Deque<JexlNode> stack = new ArrayDeque<>();
+        stack.push(node);
+        
+        boolean allIdentifiers = true;
+        
+        // iterative depth-first traversal of tree to avoid stack
+        // overflow when traversing large or'd lists
+        JexlNode current;
+        JexlNode child;
+        while (!stack.isEmpty()) {
+            current = stack.pop();
+            
+            if (current instanceof ASTOrNode) {
+                for (int i = current.jjtGetNumChildren() - 1; i >= 0; i--) {
+                    child = JexlASTHelper.dereference(current.jjtGetChild(i));
+                    stack.push(child);
+                }
+            } else {
+                children.push(current);
+                if (allIdentifiers && !(current instanceof ASTIdentifier)) {
+                    allIdentifiers = false;
+                }
+            }
+        }
+        
+        // If all ASTIdentifiers, then traverse the whole queue. Otherwise, we can attempt to short circuit.
+        State result = null;
+        if (allIdentifiers) {
+            // Likely within a function and must visit every child in the stack.
+            // Failure to do so will short circuit value aggregation leading to incorrect function evaluation.
+            while (!children.isEmpty()) {
+                // Child nodes were put onto the stack left to right. PollLast to evaluate left to right.
+                result = (State) interpretOr(children.pollLast().jjtAccept(this, data), result);
+            }
+        } else {
+            
+            // We are likely within a normal union and can short circuit
+            while (!children.isEmpty()) {
+                // Child nodes were put onto the stack left to right. PollLast to evaluate left to right.
+                result = (State) interpretOr(children.pollLast().jjtAccept(this, data), result);
+                
+                if (result.matched && !result.fromUnknown) {
+                    break;
```

Comment:
if even one component of the children is fromUnknown this will break won't it? You may not have processed the child that has this state yet...

Reply:
This behavior is intended. If the choice exists between `(FIELD || INCOMPLETE_FIELD)` then the interpreter uses the least-intensive option.